packages/edit-site/src/components/page-actions/delete-page-menu-item.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { useState } from '@wordpress/element';
+import { store as coreStore } from '@wordpress/core-data';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	MenuItem,
+	__experimentalConfirmDialog as ConfirmDialog,
+} from '@wordpress/components';
+import { store as noticesStore } from '@wordpress/notices';
+
+export default function DeletePageMenuItem( { postId, onRemove } ) {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+	const { deleteEntityRecord } = useDispatch( coreStore );
+	const page = useSelect(
+		( select ) =>
+			select( coreStore ).getEntityRecord( 'postType', 'page', postId ),
+		[ postId ]
+	);
+	async function removePage() {
+		try {
+			await deleteEntityRecord( 'postType', 'page', postId );
```

Comment:
The `deleteEntityRecord` doesn't throw an error. You'll have to use `getLastEntityDeleteError` selector.

```js
const { deleteEntityRecord } = useDispatch( coreStore );

/*...*/

const lastError = getLastEntityDeleteError( 'postType', 'page', postId );

if ( lastError ) {
    throw lastError;
}
```

P.S. Don't forget to set `isModalOpen` state to false at the end of the callback.

Reply:
Yes, that will work as well. I just forgot about the new flag 😅

Reply:
The snippet above seems to work well.

Reply:
The last remaining item would be to close the confirmation dialog at the end of the `removePage`; otherwise, it will remain open.

Reply:
Done [here](https://github.com/WordPress/gutenberg/pull/51101/commits/0322b346ad40404352cf0710d482ce4d4a67e21f). Thanks George!